### ISSUE TITLE
Upgrade Ruby & Rails

### ISSUE THREAD

**Issue by subelsky** [coyote-team/coyote#118]
Ideally get us all the way to Ruby 2.4.1 and Rails 5.1.2

**Comment by subelsky** [comment]
Getting stuck here due to strange issue where scopes on `ApplicationRecord` objects are not getting loaded - so stuff like `Image.unassigned` raises `MethodMissing`. It actually looks as if none of the class bodies get loaded.

Will return to this later or hopefully get some help, anxious to get back to feature work

**Comment by subelsky** [comment]
I had a good idea right after I commented this morning. This ticket is complete. I ended up needing to update most of the other gems to make the dependencies work. 

The build is breaking on Travis CI but that looks like a database issue a la https://github.com/travis-ci/travis-ci/issues/8087 since our database now dumps a Postgres 9.6.3-specific option that messes up Travis' default Postgres install (9.3); see the [build failure](https://travis-ci.org/coyote-team/coyote/builds/261057255) for more. I'm hoping this will get fixed on Travis' end, otherwise I will revisit this in a few days.
